File: May-18/23-05-18/c++/01-Sumits-String.cpp
#include<bits/stdc++.h>

using namespace std;

int main()
{
    int t;
    cin >> t;
    while(t--)
    {
        string str;
        cin >> str;
        int n = str.length();
        int val = str[0] - 'a';
        bool flag = true;
        for(int i = 1; i < n; i++)
        {
            int val2 = str[i] - 'a';
            if(abs(val2 - val) != 1 and abs(val2 - val) != 25)
            {
                flag = false;
                break;
            }
            val = val2;
        }
        if(flag)
            cout << "YES\n";
        else
            cout << "NO\n";
    }
}

File: May-18/23-05-18/python/01-Sumits-String.py
for _ in range(int(input())):
    data = input()
    n = len(data)
    val = ord(data[0]) - ord('a')
    flag = True
    for i in range(1, n):
        val2 = ord(data[i]) - ord('a')
        tmp = abs(val2 - val)
        
        if(tmp != 1 and tmp != 25):
            flag = False
            break
        val = val2
    if(flag == True):
        print("YES")
    else:
        print("NO")
